pkg/componentreadiness/jiraintegrator/jiraintegrator.go

```diff
@@ -0,0 +1,524 @@
+		return fmt.Sprintf("Probability of significant improvement: %.2f%%\n\n", (1-fisherExact)*100)
+	}
+	return "There is no significant evidence of regression\n\n"
+}
+
+func getStatsString(prefix string, stats crtype.TestDetailsReleaseStats, from, end string) string {
+	return fmt.Sprintf(prefix+" Release: %s\n"+
+		"\tStart Time: %s\n"+
+		"\tEnd Time: %s\n"+
+		"\tSuccess Rate: %.2f%%\n"+
+		"\tSuccesses: %d\n"+
+		"\tFailures: %d\n"+
+		"\tFlakes: %d\n\n",
+		stats.Release, from, end, stats.SuccessRate*100, stats.SuccessCount, stats.FailureCount, stats.FlakeCount,
+	)
+}
+
+// getExistingIssuesForComponent gets existing issues for a component based on
+// (a) have Regression label defined by jiraLabelIngregration
+// (b) has the "Affects Version/s" set to the sample version,
+// (c) were reported by the CR JIRA service account
+// Issues will be ordered by creation time
+func (j JiraIntegrator) getExistingIssuesForComponent(view crtype.View, component string) ([]jira.Issue, error) {
+	searchOptions := jira.SearchOptions{
+		MaxResults: 1,
+		Fields: []string{
+			"key",
+			"status",
+			"resolutiondate",
+			jiraCustomFieldReleaseBlockerName,
+			"unknowns",
+		},
+	}
+	jqlQuery := fmt.Sprintf("project=%s&&component='%s'&&creator=%s&&affectedVersion=%s&&labels in (%s) ORDER BY createdDate",
+		jiraProjectKey, component, jiraCreator, view.SampleRelease.Release, jiraLabelIngregration)
+	issues, _, err := j.jiraClient.SearchWithContext(context.Background(), jqlQuery, &searchOptions)
+	return issues, err
+}
+
+func (j JiraIntegrator) isPreRelease(release string) bool {
+	for _, r := range j.releases {
+		if r.Release == release {
+			if r.Status == releaseStatusDevelopment {
+				return true
+			}
+			break
+		}
+	}
+	return false
+}
+
+func isReleaseBlockerApproved(existing *jira.Issue) bool {
+	type releaseBlockerField struct {
+		Self     string `json:"self"`
+		ID       string `json:"id"`
+		Disabled bool   `json:"disabled"`
+		Value    string `json:"value"`
+	}
+	var obj *releaseBlockerField
+	err := jirautil.GetUnknownField(jiraCustomFieldReleaseBlockerName, existing, func() interface{} {
+		obj = &releaseBlockerField{}
+		return obj
+	})
+	if err == nil && obj.Value == jiraCustomFieldReleaseBlockerValueApproved {
+		return true
+	}
+	return false
+}
+
+// updateExistingJiraIssue updates existing issue by
+// a. adding a new comment containing a CR link with the most recently analyzed time window where the regression is still manifesting.
+// b. if pre-release, label the ticket as a Release Blocker if someone removed it
+func (j JiraIntegrator) updateExistingJiraIssue(view crtype.View, existing *jira.Issue) error {
+	absUrl, _, err := j.getComponentReadinessURLsForView(view)
+	if err != nil {
+		return err
+	}
+	comment := fmt.Sprintf(`This bug is still seen in component readiness. Here is [the current link|%s] for your convenience`, absUrl)
+
+	_, err = j.jiraClient.AddComment(existing.ID, &jira.Comment{Body: comment})
```

Comment:
It would be cool if this listed all the tests regressed (and their variant details) for that component each time.

Reply:
I don't remember what we said about this.

Reply:
We are going to keep daily update with summary. Will collect feedback when this is deployed and tested by a team.